File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.ts
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums: number[], k: number): number[] {
        const map: Map<number, number> = new Map();

        for (let i = 0; i < nums.length; i++) {
            map.set(nums[i], (map.get(nums[i]) || 0) + 1);
        }

        const arr = Array.from(map.entries());

        arr.sort((a, b) => b[1] - a[1]);

        const ans: number[] = [];

        for (let i = 0; i < k; i++) {
            ans.push(arr[i][0]);
        }

        return ans;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numset = set(nums)
        ans = 0

        for num in numset:
            if (num - 1) not in numset:
                length = 1
                while (num + length) in numset:
                    length += 1
                ans = max(ans, length)
        return ans
        